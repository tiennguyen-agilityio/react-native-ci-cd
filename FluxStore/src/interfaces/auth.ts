import {User} from './user';

export interface AuthKey {
  auth_key: string;
  refresh_key: string;
}

export interface AuthResponse {
  status: boolean;
  data: {
    user: User;
  };
}

export interface LoginPayLoad {
  email: string;
  password: string;
}
