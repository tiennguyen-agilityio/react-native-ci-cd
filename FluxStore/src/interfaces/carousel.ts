export interface CarouselCard {
  id?: string;
  title?: string;
  description?: string;
  image: string;
}
