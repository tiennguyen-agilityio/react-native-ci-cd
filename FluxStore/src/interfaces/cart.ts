interface ProductItem {
  id: string;
  image: string;
  name: string;
  price: number;
}

export interface Cart {
  id: string;
  product: ProductItem;
  sizes: string;
  colors: string;
  quantity: number;
  isChecked: boolean;
}
