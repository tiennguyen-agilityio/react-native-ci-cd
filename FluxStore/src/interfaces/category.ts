export interface Category {
  id: string;
  label: string;
  key: string;
  Icon: Element;
}
