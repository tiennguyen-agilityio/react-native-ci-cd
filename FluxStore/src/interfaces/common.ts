import {AxiosRequestConfig} from 'axios';

export interface AxiosResponse<T = object> {
  data: T;
  status: number;
  statusText: string;
  headers: object;
  config: AxiosRequestConfig;
  request?: object;
}

export interface ApiErrorResponse {
  message: string;
  code: string;
}

export interface Result<T> {
  status?: boolean;
  data?: T;
  error?: string;
}
