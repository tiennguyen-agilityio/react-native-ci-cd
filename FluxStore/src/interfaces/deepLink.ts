export interface DeepLinkData {
  isProductList?: boolean;
  id?: string;
}
