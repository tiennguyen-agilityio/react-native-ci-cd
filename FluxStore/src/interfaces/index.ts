export * from './common';
export * from './theme';
export * from './navigation';
export * from './product';
export * from './review';
export * from './cart';
export * from './category';
export * from './carousel';
export * from './auth';
export * from './user';
export * from './deepLink';
