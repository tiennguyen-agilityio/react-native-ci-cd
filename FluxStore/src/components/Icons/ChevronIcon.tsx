import {memo, useMemo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {useThemeStore} from '@/stores';
import {DIRECTION, IconProps} from '@/interfaces';
import Animated, {useAnimatedStyle, withTiming} from 'react-native-reanimated';

interface ChevronIconProps extends IconProps {
  direction?: DIRECTION;
  rotate?: number;
  duration?: number;
}

export const ChevronIcon = memo(
  ({
    width = 14,
    height = 9,
    color,
    direction = DIRECTION.RIGHT,
    rotate,
    duration = 300,
    disabled = false,
    style,
    onPress,
  }: ChevronIconProps) => {
    const {
      theme: {text},
    } = useThemeStore();

    const rotateValue = useMemo((): number => {
      switch (direction) {
        case DIRECTION.DOWN:
          return 0;

        case DIRECTION.LEFT:
          return 90;

        case DIRECTION.UP:
          return 180;

        case DIRECTION.RIGHT:
        default:
          return 270;
      }
    }, [direction]);

    const animatedStyle = useAnimatedStyle(() => ({
      transform: [
        {
          rotate: withTiming(`${rotate ?? rotateValue}deg`, {duration}),
        },
      ],
    }));

    return (
      <TouchableOpacity
        testID="chevron-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Animated.View style={[animatedStyle, {width, height}]}>
          <Svg width={width} height={height} viewBox="0 0 14 9" fill="none">
            <Path d="M13 1L7 7 1 1" stroke={color || text.primary} strokeWidth={2} />
          </Svg>
        </Animated.View>
      </TouchableOpacity>
    );
  },
);
