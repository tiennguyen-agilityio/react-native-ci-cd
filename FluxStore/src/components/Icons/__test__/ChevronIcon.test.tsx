import {render, fireEvent, screen} from '@testing-library/react-native';
import {ChevronIcon} from '../ChevronIcon'; // adjust the path as needed
import {DIRECTION} from '@/interfaces';

describe('ChevronIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<ChevronIcon />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with direction Left', () => {
    const {toJSON} = render(<ChevronIcon direction={DIRECTION.LEFT} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with direction up', () => {
    const {toJSON} = render(<ChevronIcon direction={DIRECTION.UP} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with direction down', () => {
    const {toJSON} = render(<ChevronIcon direction={DIRECTION.DOWN} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when pressed and not disabled', () => {
    render(<ChevronIcon onPress={onPress} />);
    fireEvent.press(screen.getByTestId('chevron-icon'));
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when disabled true', () => {
    render(<ChevronIcon onPress={onPress} disabled />);
    fireEvent.press(screen.getByTestId('chevron-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });

  it('applies correct rotation based on direction', () => {
    const {toJSON} = render(<ChevronIcon direction={DIRECTION.UP} />);
    const json = toJSON();
    expect(json).toBeTruthy();
  });
});
