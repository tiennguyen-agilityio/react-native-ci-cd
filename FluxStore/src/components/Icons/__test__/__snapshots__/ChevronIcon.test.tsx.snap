// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="chevron-icon"
>
  <View
    style={
      [
        {
          "transform": [
            {
              "rotate": "270deg",
            },
          ],
        },
        {
          "height": 9,
          "width": 14,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={9}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={9}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 9,
            "width": 14,
          },
        ]
      }
      vbHeight={9}
      vbWidth={14}
      width={14}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13 1L7 7 1 1"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4280098594,
              "type": 0,
            }
          }
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ChevronIcon renders correctly (matches snapshot) with direction Left 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="chevron-icon"
>
  <View
    style={
      [
        {
          "transform": [
            {
              "rotate": "90deg",
            },
          ],
        },
        {
          "height": 9,
          "width": 14,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={9}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={9}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 9,
            "width": 14,
          },
        ]
      }
      vbHeight={9}
      vbWidth={14}
      width={14}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13 1L7 7 1 1"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4280098594,
              "type": 0,
            }
          }
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ChevronIcon renders correctly (matches snapshot) with direction down 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="chevron-icon"
>
  <View
    style={
      [
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        },
        {
          "height": 9,
          "width": 14,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={9}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={9}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 9,
            "width": 14,
          },
        ]
      }
      vbHeight={9}
      vbWidth={14}
      width={14}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13 1L7 7 1 1"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4280098594,
              "type": 0,
            }
          }
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ChevronIcon renders correctly (matches snapshot) with direction up 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="chevron-icon"
>
  <View
    style={
      [
        {
          "transform": [
            {
              "rotate": "180deg",
            },
          ],
        },
        {
          "height": 9,
          "width": 14,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={9}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={9}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 9,
            "width": 14,
          },
        ]
      }
      vbHeight={9}
      vbWidth={14}
      width={14}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13 1L7 7 1 1"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4280098594,
              "type": 0,
            }
          }
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
